### PR TITLE
Api rr cleanup

### PR DESCRIPTION
**Pull request by tacaswell**
there is more to come, but might be worth looking at and merging this as-is before I make it worse.

### CODE REVIEW

**Comment by danielballan** [comment]
Judging from the Travis tests it looks like some py3-isms may have sneaked in.

File: databroker/resource_registry/base.py [hunk 1]
```diff
@@ -251,9 +251,6 @@ def get_spec_handler(self, resource):
         root = resource.get('root', '')
         root = self.root_map.get(root, root)
         if root:
-            if os.path.isabs(rpath):
```
Comment (danielballan) [inline databroker/resource_registry/base.py:232]: :+1:

**Comment by tacaswell** [comment]
This is probably ready for a commit-by-commit review, but I am concerned by the coverage drop on `db.hs.core`

**Comment by danielballan** [comment]
When the dust settles, the end of this week might be a good time to sprint on vertical integration tests.

**Pull request merged**